tests/api/test_api_v2.py

```diff
@@ -161,6 +162,69 @@ def test_get_component_analyses_with_result_not_none(
         self.assertEqual(analyses_result, result)
 
 
+@pytest.mark.usefixtures('client_class')
+class TestCAPostApi(unittest.TestCase):
+    """Component Analyses Unit Tests."""
+
+    @patch('bayesian.api.api_v2.known_package_flow')
+    @patch('bayesian.api.api_v2.unknown_package_flow')
+    @patch('bayesian.api.api_v2.get_ca_batch_response')
+    def test_get_component_analyses_post(self, _mock1, _mock2, _mock3):
```

Comment:
These kinda testing pattern would hamper the refactoring. 

Test input and expectation remains constant regardless of the function name. I highly encourage to mock only the external resources access(in this case graph DB access) and always stick with endpoint/route handler as a unit test entry point.

Reply:
Disagree.

Reply:
Can you elaborate why do you disagree with this?

Reply:
There is a difference b/w Unit Test and Integration Tests and what you are suggesting is a part of Integration Testing
https://www.guru99.com/unit-test-vs-integration-test.html

Reply:
Unit testing means testing components in isolation. It doesn't limit the component scope to function.

> Unit example
> Let's take an example - you've got an Order, which can have many OrderLines, a ShippingAddress and a Customer.
> 
> Do we have 4 units here, representing each class? It depends, but most likely it may be easier to treat the whole thing as a Unit. You can write a test which test the whole thing through the Order object. The test is never aware of the existence of the ShippingAddress. It's an internal implementation detail of the Order unit.
> 
> A class doesn't usually make a good Unit, it's usually a collection of classes that is interesting.
> 
> This way of defining a Unit gives you tests, that don't need to change whenever the internals change - that's a good thing. You don't want to change tests on every refactoring - in fact it's a smell.
> 
Reference: https://andrzejonsoftware.blogspot.com/2014/04/tdd-and-rails-what-makes-good-unit.html

Also I would recommend to go through https://medium.com/@Cyrdup/unit-testing-youre-doing-it-wrong-407a07692989